File: generated_code/python/solution_2265_1.py
from typing import List

class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        # Result array initialized with zeros of the same size
        n = len(nums)
        ans = [0] * n
        
        # We can solve this optimally in a single pass using two pointers.
        # 'left' tracks where the next element smaller than 'pivot' should go.
        # 'right' tracks where the next element greater than 'pivot' should go (from the end).
        left = 0
        right = n - 1
        
        # In a single pass, we can place elements smaller than pivot from the left,
        # and elements greater than pivot from the right.
        # To maintain stable relative ordering for elements greater than pivot, 
        # we iterate forward for 'left' elements and backward for 'right' elements.
        for i in range(n):
            if nums[i] < pivot:
                ans[left] = nums[i]
                left += 1
            if nums[n - 1 - i] > pivot:
                ans[right] = nums[n - 1 - i]
                right -= 1
                
        # Fill the remaining middle section with elements equal to pivot
        while left <= right:
            ans[left] = pivot
            left += 1
            
        return ans

File: generated_code/cpp/solution_2265_1.cpp
#include <vector>

using namespace std;

class Solution {
public:
    // Time Complexity: O(N) where N is the number of elements in nums. 
    // We iterate through the array twice (or once with a two-pointer trick). Here, we use a 
    // single-pass stable partition approach with an auxiliary result vector for clarity and O(N) space.
    // Space Complexity: O(N) to store the rearranged elements in the result vector.
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int n = nums.size();
        vector<int> result(n);
        
        int left = 0;
        int right = n - 1;
        
        // Two-pointer approach to fill smaller elements from the left and 
        // larger elements from the right to maintain their respective relative order.
        for (int i = 0, j = n - 1; i < n; ++i, --j) {
            if (nums[i] < pivot) {
                result[left++] = nums[i];
            }
            if (nums[j] > pivot) {
                result[right--] = nums[j];
            }
        }
        
        // Fill the remaining spots between 'left' and 'right' with the pivot value.
        while (left <= right) {
            result[left++] = pivot;
        }
        
        return result;
    }
};